darks_driftwind_crypt/developmental_testing.dsc

```diff
@@ -0,0 +1,166 @@
+    after player enters region_five_name:
+      - define active <server.has_flag[region_five_name.players]>
+      - flag server region_five_name.players:->:<player>
+
+      - if <[active]>:
+        - stop
+
+      # ! fireball_speed_magnitude is the speed of the fireballs; higher integers are slow, lower integers are fast
+      - define fireball_speed_magnitude 4
+
+      # ! these are pre-defined spawn locations for the fireballs;
+      # | list can be generated and added to with: `/ex flag server region_five_name.fireball_spawn_locations:->:<player.location>` at the location you're standing at
+      # | list can be purged with `/ex flag server region_five_name.fireball_spawn_locations:!`
+      - foreach <server.flag[region_five_name.fireball_spawn_locations]>:
+        - spawn fireball save:fireball_entities
+        - flag server region_five_name.fireball_entities <entry[fireball_entities].spawned_entities>
+
+      - while <server.has_flag[region_five_name.players]>:
+        - foreach <server.flag[region_five_name.fireball_entities]> as:entity:
+          - if !<[entity].exists>:
+            - flag server region_five_name.fireball_entities:<-:<[entity]>
+            - foreach next
+          - adjust <[entity]> velocity:<player.location.sub[<[entity].location>].div[<[fireball_speed_magnitude]>]>
+          - wait 5t
+
+    after player exits region_five_name:
+      - flag server region_five_name..players:<-:<player>
+      - if !<server.has_flag[region_five_name.players]>:
+        - remove <server.flag[region_five_name.fireball_entities].filter[exists]>
+        - flag server region_five_name:!
+
+# @ #10 I also need a boundary that will force teleport the player up if they land on it on the "fog" below
+    after player enters region_six_name:
+      - run fancy_painful_teleport defmap:teleport=fog_teleport_respawn
+
+# @ 11 I need this door to open when a player is close to it
+    after player enters region_seven_name:
+      - run open_gate_task defmap:door=two
+
+# @ #12 I need a system that will give the player the book that is currently on the lectern when they read it, and remove it when they exit the room*
+#13 I also need a system that will teleport the player to the start of the room if they step on an incorrect pressure plate
+  #- The solution is on the ceiling in wool
+
+    after player right clicks lectern:
+      - if <player.has_flag[region_six.has_book]>:
+        - stop
+      - flag player region_six.has_book
+      - give <context.location.inventory.list_contents.first>
+    after player exits region_six:
+      - if <player.inventory.contains_item[written_book]>:
+        - take item:written_book
+
+    # ! bad pressure plates would just be easily defined by flagging them via `/ex flag <player.cursor_on> bad_pressure_plate`
+    after player stands on *_pressure_plate:
+      - if <context.location.has_flag[bad_pressure_plate]>
+      - run fancy_painful_teleport defmap:teleport=region_six_teleport_respawn
```

Comment:
this event would fail to run, whoops.
```suggestion
      - if <context.location.has_flag[bad_pressure_plate]>:
        - run fancy_painful_teleport defmap:teleport=region_six_teleport_respawn
```